KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/SqliteTextExtractor.java

```diff
@@ -0,0 +1,393 @@
+        logger.log(Level.WARNING, msg, exception); //NON-NLS
+    }
+
+    /**
+     * Supports only the sqlite mimetypes
+     *
+     * @param file           Content file
+     * @param detectedFormat Mimetype of content file
+     *
+     * @return true if x-sqlite3
+     */
+    @Override
+    boolean isSupported(Content file, String detectedFormat) {
+        return SQLITE_MIMETYPE.equals(detectedFormat);
+    }
+
+    /**
+     * Returns an input stream that will read from a sqlite database.
+     *
+     * @param source Content file
+     *
+     * @return An InputStream that reads from a Sqlite database.
+     *
+     * @throws
+     * org.sleuthkit.autopsy.keywordsearch.TextExtractor.TextExtractorException
+     */
+    @Override
+    public Reader getReader(Content source) throws TextExtractorException {
+        //Firewall for any content that is not an AbstractFile
+        if(!AbstractFile.class.isInstance(source)) {
+            try {
+                return CharSource.wrap(EMPTY_CHARACTER_SEQUENCE).openStream();
+            } catch (IOException ex) {
+                throw new TextExtractorException(
+                    String.format("Encountered an issue wrapping blank string" //NON-NLS
+                            + " with CharSource for non-abstract file with id: [%s]," //NON-NLS
+                            + " name: [%s].", source.getId(), source.getName())); //NON-NLS
+            }
+        }
+        
+        try (AbstractReader reader = FileReaderFactory.createReader(
+                SQLITE_MIMETYPE, (AbstractFile) source)) {
+            final CharSequence databaseContent = getDatabaseContents(source, reader);
+            //CharSource will maintain unicode strings correctly
+            return CharSource.wrap(databaseContent).openStream();
+        } catch (FileReaderInitException | IOException ex) {
+            throw new TextExtractorException(
+                    String.format("Encountered a FileReaderInitException" //NON-NLS
+                            + " when trying to initialize a SQLiteReader" //NON-NLS
+                            + " for AbstractFile with id: [%s], name: [%s].", //NON-NLS
+                            source.getId(), source.getName()));
+        }
+    }
+
+    /**
+     * Queries the sqlite database and adds all tables and rows to a
+     * TableBuilder, which formats the strings into a table view for clean
+     * results while searching for keywords in the application.
+     *
+     * @param reader Sqlite reader for the content source
+     * @param source Sqlite file source
+     */
+    private CharSequence getDatabaseContents(Content source, AbstractReader reader) {
+        try {
+            Map<String, String> tables = reader.getTableSchemas();
+            Collection<String> databaseStorage = new LinkedList<>();
+
+            Integer charactersCopied = loadDatabaseIntoCollection(databaseStorage,
+                    tables, reader, source);
```

Comment:
Any reason why 'tables' cannot be obtained from 'reader' in loadDatabaseIntoCollection()? Would be one less parameter to pass.